crates/ruff_python_parser/src/parser.rs

```diff
@@ -418,6 +425,208 @@ impl ParseErrorType {
     }
 }
 
+/// An expression that may be parenthesized.
+#[derive(Clone, Debug)]
+pub(super) struct ParenthesizedExpr {
+    /// The range of the expression, including any parentheses.
+    pub(super) range: TextRange,
+    /// The underlying expression.
+    pub(super) expr: Expr,
+}
+
+impl ParenthesizedExpr {
+    /// Returns `true` if the expression is parenthesized.
+    pub(super) fn is_parenthesized(&self) -> bool {
+        self.range.start() != self.expr.range().start()
```

Comment:
I changed the condition here to only test the start as this should be sufficient